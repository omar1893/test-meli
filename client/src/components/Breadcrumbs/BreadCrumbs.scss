@import "../../../src/index.scss";

.breadcrumbs {
  @include grid-positioning(2, 12);
  font-size: 1.4rem;
  margin-top: 1.6rem;
  margin-bottom: 1.6rem;
  display: flex;
  align-items: center;
  color: $dark-gray;
  font-weight: 300;
  text-decoration: none;

  .breadcrumbs__item {
    margin-right: 0.8rem;
    text-transform: capitalize;
  }

  .breadcrumbs__item:last-child {
    font-weight: bold;
  }
}

.breadcrumbs.empty {
  margin-bottom: 0;
}
