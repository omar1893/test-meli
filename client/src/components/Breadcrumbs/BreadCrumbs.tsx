import React from "react";
import { BreadcrumbsProps } from "../../interfaces";
import "./BreadCrumbs.scss";

const BreadCrumbs = ({ categories }: BreadcrumbsProps) => {
  return (
    <div className={`breadcrumbs ${!categories.length && 'empty'}`}>
        {categories.map((category, index) => (
            <span className="breadcrumbs__item" key={category}>{category} {index < categories.length - 1 && (<span className="breadcrumbs__item--separator">{">"}</span>)}</span>
        ))}
    </div>
  );
};

export default BreadCrumbs;
