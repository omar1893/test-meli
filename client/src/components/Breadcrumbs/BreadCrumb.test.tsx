import React from 'react';
import { render, screen } from '@testing-library/react';
import BreadCrumbs from './BreadCrumbs';

describe('BreadCrumbs', () => {
  it('Should match snapshot', () => {
    const categories = ['Category 1', 'Category 2', 'Category 3'];
    const { container } = render(<BreadCrumbs categories={categories} />);
    expect(container).toMatchSnapshot();
  });

  it('Should render categories correctly', () => {
    const categories = ['Category 1', 'Category 2', 'Category 3'];
    render(<BreadCrumbs categories={categories} />);
  
    categories.forEach(category => {
      const categoryElement = screen.getByText(category);
      expect(categoryElement).toBeInTheDocument();
    });
  });

  it('Should render ">" between categories', () => {
    const categories = ['Category 1', 'Category 2', 'Category 3'];
    const { container } = render(<BreadCrumbs categories={categories} />);
    const separators = container.querySelectorAll('.breadcrumbs__item--separator');

    expect(separators.length).toBe(categories.length - 1);
    separators.forEach(separator => {
      expect(separator.textContent).toBe('>');
    });
  });

  it('Should render empty class when categories are empty', () => {
    const categories: any = [];
    const { container } = render(<BreadCrumbs categories={categories} />);
    const breadcrumbsElement = container.querySelector('.breadcrumbs');
    expect(breadcrumbsElement).toHaveClass('empty');
  });
});