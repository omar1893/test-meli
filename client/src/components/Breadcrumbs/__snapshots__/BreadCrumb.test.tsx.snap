// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs Should match snapshot 1`] = `
<div>
  <div
    class="breadcrumbs false"
  >
    <span
      class="breadcrumbs__item"
    >
      Category 1
       
      <span
        class="breadcrumbs__item--separator"
      >
        &gt;
      </span>
    </span>
    <span
      class="breadcrumbs__item"
    >
      Category 2
       
      <span
        class="breadcrumbs__item--separator"
      >
        &gt;
      </span>
    </span>
    <span
      class="breadcrumbs__item"
    >
      Category 3
       
    </span>
  </div>
</div>
`;
